src/core/xc_fallback.jl

```diff
@@ -0,0 +1,74 @@
+# TODO Temporary, until Libxc.jl is used
+@enum FunctionalFamily begin
+    family_unknown  = -1
+    family_lda      = 1
+end
+mutable struct Functional
+    name::String
+    family::FunctionalFamily
+    n_spin::Int
+end
+function Functional(identifier)
+    Functional(string(identifier), family_lda, 1)
+end
+# TODO end temporary
+
+
+function lda_x!(ρ; α=1, E=nothing, Vρ=nothing)
+    tmp = -α * cbrt.(3/π * ρ)
```

Comment:
You want a `@.` here to avoid temporaries. See comment below to get to zero allocs.

Reply:
Good point ... haven't really looked into that so far.

Reply:
That's changed now.